### PR TITLE
optimize 404 pages

### LINKED ISSUE #5080: search button throws an error on on 404 page
image

Error is thrown when you tap on the magnifying glass icon.

Strangely, _doesn't_ occur on https://amp.dev/404, but does on any other page that is a 404

### PR DESCRIPTION
**Pull request by patrickkettner**
probably a bad idea, but fixes #5080

### CODE REVIEW

File: platform/lib/routers/notFound.js [hunk 1]
```diff
@@ -18,13 +18,21 @@
...
+
+const optimizer = AmpOptimizer.create();
 
 // eslint-disable-next-line no-unused-vars
 module.exports = (req, res, next) => {
   setMaxAge(res, 60 * 10); // ten minutes
   if (req.headers.accept && req.headers.accept.includes('text/html')) {
-    res.status(404).sendFile('404.html', {root: pagePath()});
-    return;
+    optimizer.transformHtml(FourOhFour).then((optimized) => {
```
Comment (sebastianbenz) [inline platform/lib/routers/notFound.js:32]: few things:

1. please cache the optimized page in memory
2. Ideally, we'd be sharing the optimizer config with growPages.js
3. please use async await.

Comment (patrickkettner) [reply platform/lib/routers/notFound.js:32]: thanks! I wasn't sure if optimizing the 404 page is the actual "correct" thing, as opposed to just including the `<amp-lightbox>` code in the search partial.
Will update with your requests

Comment (patrickkettner) [reply platform/lib/routers/notFound.js]: updated

**Review by sebastianbenz — APPROVED** [review]
thanks!

**Pull request merged**